### ISSUE TITLE
MimeType "application/json" missing from gzip configuration

### ISSUE THREAD

**Issue by davedawkins** [SaturnFramework/Saturn#270]
Somewhere between 0.11 and 0.14, `use_gzip` was modified to add this code:

```
          let additionalMime = [|
            "application/x-yaml";
            "image/svg+xml";
            "application/octet-stream";
            "application/x-font-ttf";
            "application/x-font-opentype";
            "application/x-javascript";
            "text/javascript";
          |]
          o.MimeTypes <- if not (isNull o.MimeTypes) then  Seq.append o.MimeTypes additionalMime else Seq.ofArray (additionalMime)
```
The expectation seems to be that these would be added to defaults (such as "application/json"), but from what I can tell, it seems to be the definitive list of gzippable mime types.

So, in v0.11, `o.MimeTypes` is not set in `use_gzip`, and a response of type `application/json` gets gzipped just fine.

In 0.14, that same response will not be gzipped, *unless* we add `application/json` into the `additionalMime` array.

Disclaimer: my research is based on pulling this code into a Giraffe app. I'm not using Saturn. I did however take the `safe` template and create a `my_use_gzip` to test this with.

(Out of interest the safe-stack template uses Saturn 0.11.0.)

Hopefully this all makes sense. 

My proposed fix is to add `application/json` into `additionalMime`. I don't know what other defaults have effectively been lost though

```
          let additionalMime = [|
            "application/x-yaml";
            "image/svg+xml";
            "application/octet-stream";
            "application/x-font-ttf";
            "application/x-font-opentype";
            "application/x-javascript";
            "text/javascript";
            "application/json";
          |]
```

**Comment by Krzysztof-Cieslak** [comment]
I believe this change was originally contributed by @Thorium - any opinions?

**Comment by Thorium** [comment]
I wanted to have an option to deliver and zip yaml-files. As long as that is possible, I'm happy with any implementation. I think in 0.11 it wasn't.

**Comment by davedawkins** [comment]
My fix is a workaround. Ideally, we want to add to the existing defaults, not replace them. I could look a little deeper and see how that might be done.

**Comment by Krzysztof-Cieslak** [comment]
Fixed by #274
